vendor/TraceKit/tracekit.js

```diff
@@ -31,13 +31,11 @@ var UNKNOWN_FUNCTION = '?';
 var ERROR_TYPES_RE = /^(?:[Uu]ncaught (?:exception: )?)?(?:((?:Eval|Internal|Range|Reference|Syntax|Type|URI|)Error): )?(.*)$/;
 
 function getLocationHref() {
-    if (typeof document === 'undefined' || typeof document.location === 'undefined')
-        return '';
+  if (document == null || document.location == null) return '';
```

Comment:
This seems problematic. What if `document` isn't defined? Won't accessing it to make a null comparison throw an exception?

Why not do:

```javascript
if (typeof document === 'undefined' || document.location == null) return '';
```

Reply:
Thanks for catching that. I completely forgot that `document` is just a reference to `window.document` and it has to obey the same rules as every other property.